src/birdtrayapp.cpp

```diff
@@ -0,0 +1,148 @@
+    parseCmdArguments(parser);
+    
+    QString morkPath = parser.value("dump-mork");
+    if (!morkPath.isEmpty()) {
+        MorkParser::dumpMorkFile(morkPath);
+        exit(1); // TODO: Why 1? Replace with return code of dumpMorkFile
+        return;
+    }
+    QString imapString = parser.value("decode");
+    if (!imapString.isEmpty()) {
+        printf("Decoded: %s\n", qPrintable(Utils::decodeIMAPutf7(imapString)));
+        exit(1); // TODO: Why 1? Replace with EXIT_SUCCESS
+        return;
+    }
+    
+    ensureSystemTrayAvailable();
+    
+    // Load settings
+    settings = new Settings(parser.isSet("debug"));
+    if (parser.isSet("reset-settings")) {
+        settings->save(); // Saving without loading will reset the values
+    } else {
+        settings->load();
+    }
+    if (!translationLoadedSuccessfully) {
+        Utils::debug("Failed to load translation for %s", qPrintable(QLocale::system().name()));
+    }
+    autoUpdater = new AutoUpdater();
+    
+    trayIcon = new TrayIcon(parser.isSet("settings"));
+}
+
+BirdtrayApp::~BirdtrayApp() {
+    delete trayIcon;
+    delete autoUpdater;
+    delete settings;
+}
+
+BirdtrayApp* BirdtrayApp::get() {
+    return dynamic_cast<BirdtrayApp*>(QCoreApplication::instance());
+}
+
+Settings* BirdtrayApp::getSettings() const {
+    return settings;
+}
+
+AutoUpdater* BirdtrayApp::getAutoUpdater() const {
+    return autoUpdater;
+}
+
+TrayIcon* BirdtrayApp::getTrayIcon() const {
+    return trayIcon;
+}
+
+bool BirdtrayApp::event(QEvent* event) {
+    if (event->type() == QEvent::LocaleChange) {
+        if (!loadTranslations()) {
+            Utils::debug("Failed to load translation for %s", qPrintable(QLocale::system().name()));
+        }
+        return true;
+    }
+    return QApplication::event(event);
+}
+
+bool BirdtrayApp::loadTranslations() {
+    QString translationDir = QCoreApplication::applicationDirPath() + "/translations";
+    QLocale locale = QLocale::system();
+    bool success = loadTranslation(
+            qtTranslator, locale, "qt", QLibraryInfo::location(QLibraryInfo::TranslationsPath));
+    success &= loadTranslation(dynamicTranslator, locale, "dynamic", translationDir);
+    success &= loadTranslation(mainTranslator, locale, "main", translationDir);
+    return success;
+}
+
+bool BirdtrayApp::loadTranslation(QTranslator &translator, QLocale &locale,
+                                  const QString &translationName, const QString &path) {
+    if (translator.load(locale, translationName, "_", path)) {
+        return true;
+    }
+    // On Ubuntu, when switching to another language, the LANGUAGE environment variable
+    // does not include the base language, only the language with country,
+    // e.g. LANGUAGE=de_DE:en_US:en
+    // instead of LANGUAGE=de_DE:de:en_US:en
+    // As a result, the translator does not find the <translationName>_de.qm files.
+    // That's why we try to load the translation without the country appendix.
```

Comment:
This was necessary on my Vm with Ubuntu 18.04 LTS. Can you verify that this also works on your Linux distro?